docs/guide/sending-email.md

```diff
@@ -0,0 +1,226 @@
+    nodemailer: mg(auth);
+  }
+};
+
+```
+<template v-slot:caption>
+/modules/@apostrophecms/email
+</template>
+
+</AposCodeBlock>
+
+## Sending email from a module
+Once the `@apostrophecms/email` module is configured, email can be sent from any module using the `self.email(req, template, data, options)` method.
+
+The first parameter passed to this method is the page `req`.
+
+The next parameter, `template`, takes the name of a Nunjucks template that will make up the body of the email. This template should be located in the `views` template of the module. The method will pass this template, as well as an automatically generated version, to the `options` parameter of the `nodemailer` app.
+
+The `data` parameter takes an object that will be passed to the Nunjucks template for populating any customized fields. It can be accessed through `data.property` within the template.
+
+The final parameter, `options`, should be an object that contains the information for the email header. This is typically `from`, `to`, and `subject`. Any of these can have defaults set in the `options` of the `@apostrophecms/email` module, just like the transport. Any parameters specified in an individual module will override those set in this manner.
+
+### Example usage
+
+<AposCodeBlock>
+
+```js
+module.exports = {
+  extend: '@apostrophecms/piece-type',
+  options: {
+    lable: 'article',
+    pluralLabel: 'articles'
+  },
+  ...,
+  handlers(self) {
+    return {
+      'afterSave': {
+        async sendEmail(req, data) {
+          const options = {
+            from: 'admin@mysite.com',
+            to: 'editors@mysite.com',
+            subject: 'New Article added'
+          };
+          const template = 'email.html';
+          try {
+            await self.email(req, template, data, options);
+          return null;
+          } catch (err) {
+            self.apos.util.error('email notification error: ', err);
+          }
+        }
+      }
+    }
+  }
+};
+```
+
+<template v-slot:caption>
+/modules/article/index.js
+</template>
+
+</AposCodeBlock>
+
+
+<AposCodeBlock>
+
+```django
+<h1>A new article has been added to the site</h1>
+<p>Here is the blurb</p>
+{% data.title %}
+{% area data, 'blurb'}
+```
+
+<template v-slot:caption>
+/modules/article/views/email.html
+</template>
+
+</AposCodeBlock>
+
+In this example, we are creating a custom piece type that implements an article. The `handlers()` function is being passed the `afterSave` server event. This event is emitted any time this custom module saves a new article. We are using this signal to send out an email to the site editors. The data coming from the page is passed into the `email.html` template through the `data` argument to add the title and blurb for the editors to review.
+
+::: note
+Note that in the Nunjucks template, the area is passed in through the `data` context. It isn't being passed in through the `data.piece` or `data.page` context as it would for a normal Nunjucks template view.
```

Comment:
Yeah, true. Or you could take my note on the name of the parameter being `piece`, and write `{ data: { piece } }` here so that the template syntax can be what developers are used to.

Reply:
Changed and note deleted.